web/src/admin/policies/geoip/GeoIPPolicyForm.ts

```diff
@@ -0,0 +1,170 @@
+import { BasePolicyForm } from "@goauthentik/admin/policies/BasePolicyForm";
+import { DEFAULT_CONFIG } from "@goauthentik/common/api/config";
+import { first } from "@goauthentik/common/utils";
+import "@goauthentik/elements/forms/FormGroup";
+import "@goauthentik/elements/forms/HorizontalFormElement";
+import "@goauthentik/elements/forms/SearchSelect";
+
+import { msg } from "@lit/localize";
+import { TemplateResult, html } from "lit";
+import { customElement } from "lit/decorators.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+
+import { App, GeoIPPolicy, PoliciesApi, TypeCreate } from "@goauthentik/api";
+
+@customElement("ak-policy-geoip-form")
+export class GeoIPPolicyForm extends BasePolicyForm<GeoIPPolicy> {
+    loadInstance(pk: string): Promise<GeoIPPolicy> {
+        return new PoliciesApi(DEFAULT_CONFIG).policiesGeoipRetrieve({
+            policyUuid: pk,
+        });
+    }
+
+    async send(data: GeoIPPolicy): Promise<GeoIPPolicy> {
+        if (data.asnMode?.toString() === "") {
+            data.asnMode = "block";
+        }
+
+        if (data.asnList?.toString() === "") {
+            data.asnList = [];
+        } else {
+            data.asnList = data.asnList.split(",").map(Number);
+        }
+
+        if (data.countryMode?.toString() === "") {
+            data.countryMode = "block";
+        }
+
+        if (data.countryList?.toString() === "") {
+            data.countryList = [];
+        } else {
+            data.countryList = data.countryList.split(",");
+        }
```

Comment:
Defaults should be part of the back-end. Also some botched types here.

Reply:
Yeah, I'm looking at this and thinking that I'm seeing a lot of mixed messages compared to the types generated from the schema.  You shouldn't have to be splitting anything; the API should be giving those to you, and the TypeScript definitions in GeoIPPolicy.ts say getting strings where arrays are expected shouldn't be possible).  This function is probably the wrong place to be doing this.

If you're crafting a blob to `send`, start with a default and overwrite it with what you get from the form instead.